### DISCUSSION TITLE
Where should I look/hook to generate an rtlcss version of css files that are written to the entry HTML?

### DISCUSSION THREAD

**Discussion by davidmurdoch** [webdiscus/html-bundler-webpack-plugin#61] [Q&A] [answered]
There is an rtlcss webpack plugin, https://github.com/wix-incubator/rtlcss-webpack-plugin/tree/master, that I can add as a plugin and it will generate a Right-To-Left version of any CSS files that pass through.

It works great, but it obviously doesn't hook into html-bundler-webpack-plugin, so the output HTML doesn't end up with the "rtl" stylesheet.

What I'm hoping to do is bundle HTML like this:

```
<!doctype html>
<html>
  <head>
    <meta charset="utf-8">
    <title>Title</title>
    <link rel="stylesheet" type="text/css" href="../css/index.scss" title="ltr">
  </head>
  <body>
    
  </body>
</html>
```

into this:

```
<!doctype html>
<html>
  <head>
    <meta charset="utf-8">
    <title>Title</title>
    <link rel="stylesheet" type="text/css" href="../index.css" title="ltr">
    <link rel="stylesheet" type="text/css" href="../index.rtl.css" title="rtl" disabled>
  </head>
  <body>
    
  </body>
</html>
```

is this possible with the html-bundler-webpack-plugin hooks? If so, which hooks should I look into?

**Comment by webdiscus** [comment, marked as answer]
Hello @davidmurdoch,

thank you for the very interesting question.

I have added as an example the test case [hook-beforeEmit-rtlcss](https://github.com/webdiscus/html-bundler-webpack-plugin/tree/master/test/cases/hook-beforeEmit-rtlcss).

This is the working demo plugin, see [webpack.config.js](https://github.com/webdiscus/html-bundler-webpack-plugin/blob/master/test/cases/hook-beforeEmit-rtlcss/webpack.config.js), where is implemented the functionality of the [rtlcss-webpack-plugin](https://github.com/wix-incubator/rtlcss-webpack-plugin/tree/master) using the Bundler Plugin hook - [beforeEmit](https://github.com/webdiscus/html-bundler-webpack-plugin/tree/master#hook-beforeEmit).

**Reply by davidmurdoch** [reply]
Wow! That's awesome! Thanks again!
